### PR TITLE
Fixes #38992 - Update host vmware form to PF5

### PR DESCRIPTION
**Pull request by MariaAga**
Depends on  https://github.com/theforeman/foreman/pull/10811 (packing review is because of that PR)
Tests were created with help of AI.

To test:
In the host form -> select a compute resource with vmware type -> go to the virtual machine tab
a. see the new style of the alarm that says "Please select a cluster" at the bottom of the form
b. select a cluster, the Storage section is updated to PF5

Screenshots of the new form: 

Screenshot From 2026-01-13 11-08-48
Screenshot From 2026-01-09 13-49-46
Screenshot From 2026-01-08 11-42-22

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js [hunk 1]
```diff
@@ -28,92 +41,177 @@ const Disk = ({
...
+        <FormGroup label={__('Size (GB)')}>
+          <NumberInput
+            value={sizeGb}
+            min={1}
+            className="text-vmware-size"
+            onChange={event =>
+              updateDisk('sizeGb', parseInt(event.target.value, 10) || 1)
+            }
+            onPlus={() => updateDisk('sizeGb', parseInt(sizeGb, 10) + 1)}
+            onMinus={() => updateDisk('sizeGb', parseInt(sizeGb, 10) - 1)}
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:174]: onminus is disabled when size = 1

**Comment by MariaAga** [comment]
Updated buttons to only have 1 primary as before 
image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js [hunk 2]
```diff
@@ -28,92 +41,177 @@ const Disk = ({
...
+            </FlexItem>
+            {!vmExists && (
+              <FlexItem>
+                <Button
+                  variant="danger"
+                  className="close"
+                  onClick={removeDisk}
+                  ouiaId="btn-disk-remove"
+                  aria-label="Remove disk"
+                >
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: It was not immediately obvious to me what the red button would do with it being so close to the name of the disk. It was much clearer before this change

New:
image


Old:
image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js [hunk 3]
```diff
@@ -102,36 +100,50 @@ const Controller = ({
...
+          <FormGroup>
+            <Button
+              variant="secondary"
+              className="btn-remove-controller"
+              onClick={removeController}
+              isDisabled={config.vmExists}
+              ouiaId="btn-remove-controller"
+            >
+              {__('Delete Controller')}
+            </Button>
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js:143]: With this change, the "Add volume" and "Delete controller" buttons changed places. Is that intentional?

New:
image

Old:
image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js [hunk 4]
```diff
@@ -28,92 +41,177 @@ const Disk = ({
...
+                  }}
+                  onSelect={(_ev, selectedValue) => {
+                    updateStoragePod(selectedValue);
+                  }}
+                  selected={storagePod}
+                  className="storage-pod"
+                  key="storagePodsSelect"
+                  placeholder=""
+                  isDisabled={vmExists}
+                />
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: All the dropdowns seem to have some extra padding below the last item. Is that intended?

image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js [hunk 5]
```diff
@@ -28,92 +41,177 @@ const Disk = ({
...
+        <FormGroup label={__('Disk Mode')}>
+          <SimpleSelect
+            ouiaId="select-disk-mode"
+            isDisabled={vmExists}
+            onSelect={(_ev, selectedValue) => {
+              updateDisk('mode', selectedValue);
+            }}
+            initialOptions={diskModeOptions}
+          />
+        </FormGroup>
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: Could all the fields have the same width like they did in the old implementation?

New:
image

Old:
image

**Review by adamruzicka — CHANGES_REQUESTED** [review]
Not sure if it is caused by this, but submitting the form with fields missing leads to a page with rails traceback:

New:
image

Old:
image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 5 above)
Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: Also, storage pod and data store names can be rather long, so giving them more width would be nice.

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 2 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: I think it would be best to give it text, as I didnt see the old page like that, and with a wide screen the old page makes sense. 
image

Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: +1 for text

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js (hunk 3 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js:143]: Nope, moved them back to their position

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 4 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: I think I would like to add it as a global rule to remove the extra padding, as this is a reccuring issue in every PR with PF5 select. This comes from a generic css we have that adds padding at the end of every list, and every PR just overrides it separately

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 5 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: Updated to have the selects be longer 
image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 2 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: image
Added text to the button and moved it to the right

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: Will it be better to name it "remove volume" instead of "remove disk" to match "add volume"?

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 5 above)
Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: I guess I'm fine with size being small

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 2 above)
Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: Yes, volumes all the way

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 4 above)
Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: > This comes from a generic css we have that adds padding at the end of every list

Why do we even have that? Is that a workaround for some older pf quirk that we didn't like?

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: it comes from 
`@import '~bootstrap-sass/assets/stylesheets/bootstrap/variables'; `
that has this rule
```
ul,
ol {
  margin-top: 0;
  margin-bottom: 10px;
}
```

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 5 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: do you want another field to be larger?

**Comment by MariaAga** [comment]
@adamruzicka I checked this
>Not sure if it is caused by this, but submitting the form with fields missing leads to a page with rails traceback: 

And I see a page with rails traceback even without this PR, by just opening the create host page and clicking submit

**Comment by adamruzicka** [comment]
> And I see a page with rails traceback even without this PR,

"Good". Let's not worry about it here then

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 5 above)
Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: The size field is currently large enough for values of up to 4 digits. Anything above 10tb sort of breaks

image

If we can make it larger to accommodate bigger values then that would be nice

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: Ah I see, my view of the size field never got that small for some reason, updated it to support 10 chars

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 2 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js:72]: updated, thanks

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js [hunk 6]
```diff
@@ -102,36 +99,50 @@ const Controller = ({
...
+              }
+            />
+            <Button
+              variant="secondary"
+              className="btn-remove-controller"
+              onClick={removeController}
+              isDisabled={config.vmExists}
+              ouiaId="btn-remove-controller"
+            >
+              {__('Delete Controller')}
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js]: While we're unifying the language, should we also settle on `Remove` vs `Delete`?

image

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js [hunk 7]
```diff
@@ -102,36 +99,50 @@ const Controller = ({
...
+            <SimpleSelect
+              ouiaId="select-controller-type"
+              initialOptions={controllerTypeOptions}
+              isDisabled={config.vmExists}
+              onSelect={(_ev, selection) =>
+                _updateController('type', selection)
+              }
+            />
+            <Button
+              variant="secondary"
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js]: Should this be a danger variant to be in line with volume removal?

**Review by adamruzicka — COMMENTED** [review]
I don't have anything to add apart from those last two comments

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 4 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: https://github.com/theforeman/foreman/pull/10836

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js (hunk 6 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js]: https://www.patternfly.org/patterns/actions/#deletion-vs-removal 

> Deletion: Refers to the destruction of information (for example, deleting a group, system, or a role.) Often identified by a trash icon.
> 
> Removal: Refers to removing the content/system from a particular view, without the destruction of information (for example, removing something from a group or removing a user.) Often identified by a solid minus-circle icon.

Since users can fill up the fields, and that can be counted as information, I think we should use delete.
But we probably dont need a modal warning that they are deleting somethings (as suggested in the guide), right?

Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js]: Delete sounds about right, but having a modal feels excessive. So yes, as you said

**Comment by MariaAga** [comment]
This PR is waiting for a UI review from @MariSvirik and a code review from @kmalyjur

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js [hunk 8]
```diff
@@ -28,92 +37,166 @@ const Disk = ({
...
+        <FormGroup label={__('Size (GB)')}>
+          <NumberInput
+            value={sizeGb}
+            min={1}
+            className="text-vmware-size"
+            onChange={event =>
+              updateDisk('sizeGb', parseInt(event.target.value, 10) || 1)
```
Comment (kmalyjur) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: This `|| 1` makes it impossible to clear the input since it resets to 1 immediately, so when trying to change `1` to `2`, you have to write `12` and then delete `1` (or use the +/- buttons).

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js [hunk 9]
```diff
@@ -102,36 +99,50 @@ const Controller = ({
...
-          <Button
-            className="btn-remove-controller"
-            onClick={removeController}
-            disabled={config.vmExists}
-          >
-            {__('Delete Controller')}
-          </Button>
-        </div>
+        <Form>
+          <FormGroup label={__('Create controller')}>
```
Comment (kmalyjur) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js:115]: When having more controllers, there isn't a clear separation between them. It seems like `Create controller` label is a part of the previous form, but we can probably leave that for @MariSvirik to determine.

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js [hunk 10]
```diff
@@ -28,92 +37,166 @@ const Disk = ({
...
+            </FlexItem>
+            {!vmExists && (
+              <FlexItem align={{ default: 'alignRight' }}>
+                <Button
+                  variant="danger"
+                  onClick={removeDisk}
+                  ouiaId="btn-volume-remove"
+                  aria-label="Remove volume"
+                >
+                  {__('Remove volume')}
```
Comment (kmalyjur) [inline webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: Maybe a nitpick, but 'Remove volume' has a lowercase 'v' while 'Add Controller' has an uppercase 'C'. It could be nicer to unify them.

**Review by kmalyjur — CHANGES_REQUESTED** [review]
Looks good! I'm leaving a few comments, but they are not that important.

File: webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js (hunk 10 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js]: thanks I missed them! Updated the add/remove controller buttons to use sentence case to match.

**Comment by MariaAga** [comment]
Screenshot after the latest changes: 
image

**Comment by kmalyjur** [comment]
I'd ACK it, but let's wait for @MariSvirik's comments.

**Comment by MariSvirik** [comment]
I have some questions:
1. Are you Creating or Adding controller?
2. What does controller mean and is it even a proper word to use here?
3. Is the controller dropdown selecting a controller type?
4. Is Volume a proper word to use here? Should it be “Add disk” or “disk volume”?
5. Is the disk name always Hard disk or could be changed, and if so how?
6. If it is always “hard disk” what is the purpose about having it there?

Preliminary designs (reviewed by other designers, PF5 standards): https://www.figma.com/design/qMUJFPL6BpgbmNr9vctrMF/Fixes?node-id=0-1&t=qrqR29xeCdlymOMF-1 
This may change based on the answers to the questions
Consulted with Maria Aga

**Comment by adamruzicka** [comment]
> Are you Creating or Adding controller?

That probably depends on your point of view. You're preparing a virtual machine to be created. Are you adding components to it or are you creating the components?

> What does controller mean and is it even a proper word to use here?

It is [a piece of hardware](https://en.wikipedia.org/wiki/Disk_controller) that allows storage devices to be attached to the machine and yes it is the right word.

> Is the controller dropdown selecting a controller type?

More or less, it mixes together a type (NVME, SCSI, ...) and implementation (emulated lsi logic, vmware paravirtual, ...)

> Is Volume a proper word to use here? Should it be “Add disk” or “disk volume”?

Iirc "Volume" is a proper word if we stick with the vmware nomenclature

> Is the disk name always Hard disk or could be changed, and if so how?

If I'm reading it right that is just there without any way of changing it.

> If it is always “hard disk” what is the purpose about having it there?

I can't offer anything better than "it was there before".

**Comment by dgoetz** [comment]
> > If it is always “hard disk” what is the purpose about having it there?
> 
> I can't offer anything better than "it was there before".

My VMware knowledge is perhaps not 100% up to date, but I think this is caused by Foreman's capabilities/workflow here. In VMware you can add different types of virtual hardware via the same wizard, in Foreman we have here a specific section for the hard disk only. I am not sure if we tell VMware via Foreman that all hard disks are called "Hard disk" and it will just create "Hard disk", "Hard disk 2", Hard disk 3" from this, if this is the case it will probably show these names after creation. Unfortunately no access to an environment to verify this at the moment.

**Comment by MariSvirik** [comment]
https://www.figma.com/design/qMUJFPL6BpgbmNr9vctrMF/Fixes?node-id=23-1444&t=JNWm6Od2etcYV3hL-0
Screenshot 2026-03-30 at 13 51 10

**Comment by MariaAga** [comment]
It seems that having two disk with different storage pod/ data store is not supported. So I'll move those fields out of the individual disk forms.

**Comment by ekohl** [comment]
> It seems that having two disk with different storage pod/ data store is not supported. So I'll move those fields out of the individual disk forms.

My understanding of the code is that each volume belongs to a controller. Fog's interface of mapping those isn't all that good though.

Each volume should have the controller key (https://github.com/fog/fog-vsphere/blob/5971c8b4793765d50976a877471b28b07fcc2bc3/lib/fog/vsphere/requests/compute/create_vm.rb#L311). How you determine that exact key isn't clear to me. There is some code that automatically assigns it to the first one (https://github.com/fog/fog-vsphere/blob/5971c8b4793765d50976a877471b28b07fcc2bc3/lib/fog/vsphere/models/compute/server.rb#L363-L365).

**Comment by MariaAga** [comment]
> It seems that having two disk with different storage pod/ data store is not supported. So I'll move those fields out of the individual disk forms.
My understanding of the code is that each volume belongs to a controller. Fog's interface of mapping those isn't all that good though. 

@ekohl 
When I create 1 vm, with 1 controller, and 3 volumes with different storage pod/ data store, they end up being created with only one value of storage pod/ data store. 
Not sure if its a bug or a feature but I want to avoid letting a user choose fields that will be overridden anyway.

**Comment by ekohl** [comment]
Going back to the VMware level it's: A VM has 1 or more storage controllers. Each storage controller has 0 or more hard disks and/or CD/DVD-ROM devices attached to it.

The reason to add multiple controller is that you can attach a maximum number of devices to it. Those limits feel a bit arbitrary. Looking at https://techdocs.broadcom.com/us/en/vmware-cis/vsphere/vsphere/8-0/vsphere-virtual-machine-administration/configuring-virtual-machine-hardwarevsphere-vm-admin/scsi-controller-configurationvsphere-vm-admin.html you can see:

* A VM can have a maximum of 4 SCSI controllers and 4 SATA controllers. If I read it right, alternatively it can have 4 NVMe controllers. If I read it right, that's instead of SCSI and SATA.
* SCSI controllers can have 16 disks
* SATA controllers can have 30 disks
* NVMe controllers can have 15 disks

Now our UI doesn't even come close to properly supporting this with the right limitations and likely the backend code doesn't either, but that's where the ideas about multiple controllers come from: limitations from the hypervisor.

**Comment by MariaAga** [comment]
@ekohl 
reading your comment, I'm not sure if youre supporting my proposed change: select storage pod/ data store on the controller lever instead of the disk level
or not?
Or what are you proposing?

**Comment by ekohl** [comment]
Even if our API / backend doesn't really use it, from a UI perspective I would expect a hierarchy view:
* VM
  * controller 1
    * volume 1
  * controller 2
    * volume 2

> reading your comment, I'm not sure if youre supporting my proposed change: select storage pod/ data store on the controller lever instead of the disk level

AFAIK a volume always lives on a storage pod / data store. If you set it at the controller level I'd expect that an experienced VMware admin will be confused because you're inventing something new. I'd keep it at the volume level.

Note that with vCenter you can migrate an individual disk to another storage location, it isn't done at the controller level.

**Comment by MariaAga** [comment]
> If you set it at the controller level I'd expect that an experienced VMware admin will be confused because you're inventing something new. I'd keep it at the volume level.

Isnt it also confusing to set a value in the volume level and have foreman change it without you knowing?

**Comment by ekohl** [comment]
> > If you set it at the controller level I'd expect that an experienced VMware admin will be confused because you're inventing something new. I'd keep it at the volume level.
> 
> Isnt it also confusing to set a value in the volume level and have foreman change it without you knowing?

Yes, but that could be considered a backend bug too.

**Comment by ShimShtein** [comment]
I took a longer look at the volumes and controllers handling in Foreman and Fog. From what I saw, the handling is done correctly (although a bit awkward):
VMWare API expects two properties side by side - controllers and disks. The objects in the controllers array are quite simple, the only interesting property is the key, which is an integer that is used to identify the controller. The objects in the disks array have two special properties: `controller_key` and  `unit_number`. The combination of those two defines how to emulate the access to the disk. `controller_key` should correspond to the `key` property of an existing controller and `unit_number` is a running ID for each controller. By default the first controller will get a key value of `1000`.
There is a small caveat with NVME controllers that adds complexity to the picture, because those controllers are defined separately in Fog (and I think in API). 
I see that this relationship of ids is maintained in Fog and Foreman, and no code should break it.
About the storage pods/data sources: On VMWare level, it's a disk property, not a controller one. So each disk can be "implemented" on a different storage pod, and accessed by calling a different controller from the VM. I think there is also some code that selects a proper pod for the virtual machine definition itself, but I did not find enough information about this part. 

The bottom line is that the definition that should be consumed by the server should look like:
```
{
    "controllers": [
        {
            "type": "VirtualLsiLogicController",
            "key": 1000
        },
        {
            "type": "ParaVirtualSCSIController",
            "key": 1001
        }
    ],
    "volumes": [
        {
            "thin": false,
            "name": "Hard disk",
            "mode": "persistent",
            "sizeGb": 1,
            "controllerKey": 1000,
            "storagePod": "Example-DatastoreCluster",
            "datastore": null,
            "eagerZero": false
        },
        {
            "sizeGb": 1,
            "datastore": null,
            "storagePod": "Example-DatastoreCluster",
            "thin": false,
            "eagerZero": false,
            "name": "Hard disk",
            "mode": "persistent",
            "controllerKey": 1001
        }
    ]
}
```
This creates two controllers, each one of the controllers has one disk, and both disks will use the `Example-DatastoreCluster` to store the data written to the volumes.

**Comment by MariaAga** [comment]
Thanks everyone for answering and reviewing. I sat down with Shim to talk about this PR, also I saw the vsphere was changing the Data store, and not foreman.
This is how it currently used. Differences are mostly caused by how PF implements components, and to accommodate string lengths.
Code is AI assisted 
image

**Comment by ekohl** [comment]
Just based on the screenshot I think it's correct but I hope someone has the capacity to properly test it.

**Review by kmalyjur — APPROVED** [review]
LGTM from the javascript side. I hope the Rocket team will test the edge cases more.

**Review by stejskalleos — DISMISSED** [review]
- The PR is not for the full VMware form, only the storage section -> The commit message & PR title should reflect that.
- Nitpick: I can type `-20` into the `Size (GB)` field.
- If I don't select required fields `Storage Pod *` or `Data store`, the form can be submitted. I know it's a React component inserted into the form, but it would be nice not to allow it. Anyway, not gonna block on this PR.

Can you please rebase the issue over the latest develop? (because of dc528f44fe0dfafba450ea2c02fb547e9896741c)

**Comment by ShimShtein** [comment]
Did some testing, OCP compute resource is not affected.
One nitpick: when showing the list of disks, only the first disk in a controller is expanded. Is it intentional?
image
While we are at it, the defaults are also not consistent: only the first disk is defined as "thin" by default, all others are defined as "thick" (thin checkbox is unmarked). Is it also intentional?
Otherwise LGTM.

**Comment by MariaAga** [comment]
Thanks all, I will get to the review comments this week, about 
> While we are at it, the defaults are also not consistent: only the first disk is defined as "thin" by default, all others are defined as "thick" (thin checkbox is unmarked). Is it also intentional?

This was the behaviour before, should I change it to the default being thin for all?

**Comment by MariaAga** [comment]
> The PR is not for the full VMware form, only the storage section -> The commit message & PR title should reflect that.

Updated

> Nitpick: I can type -20 into the Size (GB) field.

fixed

> If I don't select required fields Storage Pod * or Data store, the form can be submitted. I know it's a React component inserted into the form, but it would be nice not to allow it. Anyway, not gonna block on this PR.

Thats not consistent with the host form behaviour for any other field (for example: name can be empty and the submit button is enabled)

**Comment by ShimShtein** [comment]
> This was the behaviour before, should I change it to the default being thin for all?

If that was the original behavior, I think it's good enough. Let's keep it that way.

**Comment by MariaAga** [comment]
Talked to @ShimShtein, and he pointed out two issues, that are now fixed:

* Volume expansion

On edit host (vmExists), all volumes are expanded by default so existing disk settings are visible without clicking each one.
On new host, the first volume and the most recently added volume stay expanded.
New disk defaults

* New volumes no longer inherit Storage Pod or Data store from an existing disk when adding a volume or controller. Each new disk starts with empty storage fields and must be configured explicitly. (This was a leftover from a discussed logic, its not changing the original behaviour)

**Review by ShimShtein — APPROVED** [review]
LGTM

**Comment by ShimShtein** [comment]
I have tested the PR with packit build:
* Create a host with the default disk
* create the host and add more disks and controllers
* Edit host with multiple disks and see the R/O disk info
* make sure OCP-V compute resource still works as previously.
